tests/compute/test_sampler.py

```diff
@@ -155,10 +155,53 @@ def test_layer_sampler():
     _test_layer_sampler()
     _test_layer_sampler(prefetch=True)
 
+def test_nonuniform_neighbor_sampler():
+    # Construct a graph with
+    # (1) A path (0, 1, ..., 99) with weight 1
+    # (2) A bunch of random edges with weight 0.
+    edges = []
+    for i in range(99):
+        edges.append((i, i + 1))
+    for i in range(1000):
+        edge = (np.random.randint(100), np.random.randint(100))
+        if edge not in edges:
+            edges.append(edge)
+    src, dst = zip(*edges)
+    g = dgl.DGLGraph()
+    g.add_nodes(100)
+    g.add_edges(src, dst)
+    g.readonly()
+
+    g.edata['w'] = F.cat([
+        F.ones((99,), F.float64, F.cpu()),
+        F.zeros((len(edges) - 99,), F.float64, F.cpu())], 0)
+
+    # Test 1-neighbor NodeFlow with 99 as target node.
+    # The generated NodeFlow should only contain node i on layer i.
+    sampler = dgl.contrib.sampling.NeighborSampler(
+        g, 1, 1, 99, 'in', transition_prob='w', seed_nodes=[99])
+    nf = next(iter(sampler))
+
+    assert nf.num_layers == 100
+    for i in range(nf.num_layers):
+        assert nf.layer_size(i) == 1
+        assert nf.layer_parent_nid(i)[0] == i
+
+    # Test the reverse direction
+    sampler = dgl.contrib.sampling.NeighborSampler(
+        g, 1, 1, 99, 'out', transition_prob='w', seed_nodes=[0])
+    nf = next(iter(sampler))
+
+    assert nf.num_layers == 100
+    for i in range(nf.num_layers):
+        assert nf.layer_size(i) == 1
+        assert nf.layer_parent_nid(i)[0] == 99 - i
```

Comment:
is it possible to check the sample distribution?

Reply:
Could you give an example of how to do so?

Reply:
If we sample many times, the frequency of the sampled result will be close to the probability distribution, although it's really tricky to set the epsilon. Have you observed the sampled results manually?

Reply:
Yes and it looked fine, although I need to put a small delay between samples because we initialize the seed with `time(nullptr)` every time `SampleSubgraph` is called.  I plan to put a better RNG in C to address this issue as well as providing reproducibility in sampling.

Reply:
I think no one is working on it. Thanks for picking this task.

Reply:
Great! Take a look at TVM's random engine codes [here](https://github.com/dmlc/tvm/blob/cffb4fba03ea582417e2630bd163bca773756af6/src/contrib/random/mt_random_engine.cc) and [here](https://github.com/dmlc/tvm/blob/cffb4fba03ea582417e2630bd163bca773756af6/src/contrib/random/random.cc). Two things to pay attention to: (1) consider using std <random> which is multi-platform; (2) use thread local entry to handle multi-thread case. Thread local entry acts like a singleton but is thread-safe. Then we can expose API to set seed to python.